Command/ClonePagesCommand.php

```diff
@@ -0,0 +1,189 @@
+            $this->getPageManager()->save($newPage);
+
+            $pageClones[$page->getId()] = $newPage;
+
+            // Clone page blocks
+            /** @var BlockInterface[] $blocks */
+            $blocks = $this->getBlockManager()->findBy(array('page' => $page));
+            foreach ($blocks as $block) {
+                $output->writeln(sprintf(' cloning block % 4s ', $block->getId()));
+                $newBlock = clone $block;
+                $newBlock->setPage($newPage);
+                $blockClones[$block->getId()] = $newBlock;
+                $this->getBlockManager()->save($newBlock);
+            }
+        }
+
+        $output->writeln('Fixing page parents and targets');
+        foreach ($pageClones as $page) {
+            if ($page->getParent()) {
+                $output->writeln(
+                    sprintf(
+                        'new parent: % 4s - % -70s - % 4s -> % 4s',
+                        $page->getId(),
+                        $page->getTitle(),
+                        $page->getParent() ? $page->getParent()->getId() : '',
+                        $pageClones[$page->getParent()->getId()]->getId()
+                    )
+                );
+                $page->setParent($pageClones[$page->getParent()->getId()]);
+            }
+
+            if ($page->getTarget()) {
+                $output->writeln(
+                    sprintf(
+                        'new target: % 4s - % -70s - % 4s',
+                        $page->getId(),
+                        $page->getTitle(),
+                        $page->getParent() ? $page->getParent()->getId() : ''
+                    )
+                );
+                $page->setTarget($pageClones[$page->getTarget()->getId()]);
+            }
+
+            $this->getPageManager()->save($newPage, true);
+        }
+
+        $output->writeln('Fixing block parents');
+        foreach ($pageClones as $page) {
+            $blocks = $this->getBlockManager()->findBy(array('page' => $page));
+            foreach ($blocks as $block) {
+                if ($block->getParent()) {
+                    $output->writeln(
+                        sprintf(
+                            'new block parent: % 4s - % 4s',
+                            $block->getId(),
+                            $blockClones[$block->getParent()->getId()]->getId()
+                        )
+                    );
+                    $block->setParent($blockClones[$block->getParent()->getId()]);
+
+                    $this->getBlockManager()->save($block, true);
+                }
+            }
+        }
+
+        $output->writeln('<info>done!</info>');
+    }
+
+    /**
+     * Prints a list of all available sites.
+     *
+     * @param OutputInterface $output
+     */
+    private function listAllSites(OutputInterface $output)
+    {
+        $output->writeln(sprintf(' % 5s - % -30s - %s', 'ID', 'Name', 'Url'));
+
+        foreach ($this->getSiteManager()->findAll() as $site) {
```

Comment:
i would call the `findAll()` one line before and create a variable `$sites` which can be used in the foreach